### PR TITLE
Initial ALH/SoundWire support

### PR DESCRIPTION
**Pull request by plbossart**
These patches are just basic definitions that can be merged independently through the SOF tree and be pushed to Mark Brown.

### CODE REVIEW

**Comment by dbaluta** [comment]
Looks good to me. I wonder if we should care about the checkpatch.pl warnings

```
CHECK: Prefer kernel type 'u32' over 'uint32_t'
#30: FILE: include/sound/sof/dai-intel.h:188:
+	uint32_t reserved[15];
```

**Comment by plbossart** [comment]
> Looks good to me. I wonder if we should care about the checkpatch.pl warnings
> 
> ```
> CHECK: Prefer kernel type 'u32' over 'uint32_t'
> #30: FILE: include/sound/sof/dai-intel.h:188:
> +	uint32_t reserved[15];
> ```

no, this is for the IPC parts and we don't want to use u32 kernel stuff here.

**Comment by plbossart** [comment]
updated branch to add ABI MINOR change to 10 (was in @slawblauciak PR and not reflected here)

**Comment by ranj063** [comment]
@plbossart should we just squash the 2nd and 3rd commits?

File: sound/soc/sof/pcm.c [hunk 1]
```diff
@@ -675,6 +675,9 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	case SOF_DAI_INTEL_HDA:
 		/* do nothing for HDA dai_link */
 		break;
+	case SOF_DAI_INTEL_ALH:
+		/* do nothing for ALH dai_link */
```
Comment (ranj063) [inline sound/soc/sof/pcm.c:679]: @plbossart I think removing the break and allowing to fall through would look better but its just my personal choice. No need to change if you feel otw

**Comment by plbossart** [comment]
> @plbossart should we just squash the 2nd and 3rd commits?

did you mean 3-4?

**Comment by ranj063** [comment]
> > @plbossart should we just squash the 2nd and 3rd commits?
> 
> did you mean 3-4?

@plbossart yes :d

**Comment by plbossart** [comment]
> > > @plbossart should we just squash the 2nd and 3rd commits?
> > 
> > 
> > did you mean 3-4?
> 
> @plbossart yes :d

ok, indeed this was an incremental work. we added patch 4 when we realized *something* was needed.

**Pull request merged**